main.tf

```
@@ -152,5 +152,11 @@ resource "aws_iam_role_policy_attachment" "kms_policy_attachment" {
   count      = var.kms_key_arn != "" ? 1 : 0
   role       = module.lambda.iam_role_name
   policy_arn = aws_iam_policy.kms_policy[count.index].arn
-} 
+}
 
+resource "aws_lambda_function_url" "lambda_url" {
+  count              = var.enable_functionurl ? 1 : 0
```

```suggestion
  count              = var.enable && lookup(var.architecture, "function_url", false) ? 1 : 0
```
This follow the architecture pattern we've established for the other event sources and lambda triggers.